packages/client/src/sw.ts

```diff
@@ -0,0 +1,131 @@
+const CACHE_NAME = 'echo-runner-cache-v1';
+const API_CACHE_NAME = 'echo-runner-api-cache-v2';
+const API_ENDPOINTS = ['/auth/user'];
+const AUTH_API_BASE = 'https://ya-praktikum.tech/api/v2/auth';
+
+const URLS = ['/index.html', '/', '/topics', '/signin', '/signup', '/profile', '/leaderboard', '/game', '/error'];
+
+self.addEventListener('install', ((event: ExtendableEvent) => {
+  console.log('🔧 Service Worker installing...');
+  event.waitUntil(
+    Promise.all([
+      caches.open(CACHE_NAME).then((cache) => {
+        console.log('Opened static cache:', CACHE_NAME);
+        return cache.addAll(URLS);
+      }),
+      caches.open(API_CACHE_NAME).then((cache) => {
+        console.log('Opened API cache:', API_CACHE_NAME);
+        return cache;
+      }),
+    ]).catch((err) => {
+      console.log('Install error:', err);
+      throw err;
+    }),
+  );
+}) as EventListener);
+
+self.addEventListener('activate', ((event: ExtendableEvent) => {
+  console.log('activate');
+  event.waitUntil(
+    caches.keys().then((cacheNames) => {
+      console.log('activate:', cacheNames);
+      return Promise.all(
+        cacheNames
+          .filter((name) => name !== CACHE_NAME && name !== API_CACHE_NAME)
+          .map((name) => {
+            caches.delete(name);
+            console.log('delete cache', name);
```

Comment:
caches.delete возвращает Promise, но внутри map он не return-ится и Promise.all получает массив undefined.


```suggestion
.map((name) => {
  console.log('delete cache', name);
  return caches.delete(name);
})
```